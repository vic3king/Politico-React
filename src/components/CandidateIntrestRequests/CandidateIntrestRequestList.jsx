import React, { Component } from 'react';
import { notify } from 'react-notify-toast';
import CandidateIntrestRequestsCard from './CandidateIntrestRequests';
import Vote from '../../services/votes';
import Loader from '../shared/Loader/Loader';
import Interest from '../../services/interest';
import errorHandler from '../../helpers/errorHandler';

class CandidateIntrestRequestsList extends Component {
  state = {
    candidates: [],
    loading: true,
  };

  async componentDidMount() {
    const pendingCandidates = await Vote.getCandidatesByStatus();

    this.setState({
      candidates: pendingCandidates.data,
      loading: false,
    });
  }

  updateCandidatesState = candidateId => {
    const { candidates } = this.state;
    const newCandidatesList = candidates.filter(
      candidate => candidate.id !== parseInt(candidateId, 10)
    );

    this.setState({ candidates: newCandidatesList });
  };

  adminAcceptRequest = async event => {
    event.preventDefault();
    this.setState({ loading: true });
    const candidateId = event.target.id;
    const formDetails = { status: 'approved' };

    const acceptedReveiw = await Interest.processRequest(
      formDetails,
      candidateId
    );

    if (acceptedReveiw.status >= 400) {
      this.setState({ loading: false });
      notify.show(errorHandler(acceptedReveiw.errors.message), 'error');
    }

    if (acceptedReveiw.status === 200) {
      this.setState({ loading: false });
      notify.show('Candidate Approved');
      this.updateCandidatesState(candidateId);
    }
  };

  adminRejectRequest = async event => {
    event.preventDefault();
    this.setState({ loading: true });
    const candidateId = event.target.id;
    const formDetails = { status: 'rejected' };

    const rejectedReveiw = await Interest.processRequest(
      formDetails,
      candidateId
    );

    if (rejectedReveiw.status >= 400) {
      this.setState({ loading: false });
      notify.show(errorHandler(rejectedReveiw.errors.message), 'error');
    }

    if (rejectedReveiw.status === 200) {
      this.setState({ loading: false });
      notify.show('Candidate Rejected');
      this.updateCandidatesState(candidateId);
    }
  };

  render() {
    const { candidates, loading } = this.state;

    return (
      <React.Fragment>
        {loading && <Loader />}
        <div>
          <div className="main_page_title">
            <div className="main-grid">
              {candidates.map(candidate => (
                <CandidateIntrestRequestsCard
                  key={candidate.id}
                  id={candidate.id}
                  fullname={`${candidate.firstname} ${candidate.lastname}`}
                  office={candidate.officename}
                  party={candidate.partyname}
                  button="Accept"
                  adminAcceptRequest={this.adminAcceptRequest}
                  adminRejectRequest={this.adminRejectRequest}
                  btnClass="btn-accept"
                  btnClass1="btn-reject"
                />
              ))}
            </div>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

export default CandidateIntrestRequestsList;
