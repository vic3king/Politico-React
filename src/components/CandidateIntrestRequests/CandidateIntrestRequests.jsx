import React from 'react';
import '../../style/intrests-requests.scss';

const CandidateIntrestRequestsCard = ({
  fullname,
  party,
  office,
  adminAcceptRequest,
  id,
  adminRejectRequest,
}) => {
  return (
    <React.Fragment>
      <div className="main-title">
        <div className="image_title_bio">
          <div className="profile_image">
            <img
              className="user_pic"
              src="https://picsum.photos/200/300/?random"
              alt=""
            />
          </div>
          <div className="title_text_box">
            <p className="name">{fullname}</p>
            <p className="user_bio">Party: {party}</p>
            <p className="user_bio">Office of Intrest: {office}</p>
          </div>
        </div>
        <div className="admin_decision_on_request">
          <div className="main_button">
            <button
              type="submit"
              id={id}
              className="btn-accept"
              onClick={adminAcceptRequest}
            >
              Accept
            </button>
            <button
              type="submit"
              id={id}
              className="btn-reject"
              onClick={adminRejectRequest}
            >
              reject
            </button>
          </div>
        </div>
      </div>
    </React.Fragment>
  );
};

export default CandidateIntrestRequestsCard;
