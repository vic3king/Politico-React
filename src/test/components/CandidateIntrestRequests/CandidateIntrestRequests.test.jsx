import React from 'react';
import { shallow } from 'enzyme';
import CandidateIntrestRequests from '../../../components/CandidateIntrestRequests/CandidateIntrestRequests';

describe('CandidateIntrestRequests component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<CandidateIntrestRequests />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('section'));
    expect(wrapper.find('div'));
    expect(wrapper.find('button'));
  });
});
