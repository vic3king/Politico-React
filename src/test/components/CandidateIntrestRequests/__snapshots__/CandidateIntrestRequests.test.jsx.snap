// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CandidateIntrestRequests component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CandidateIntrestRequestsCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="main-title"
      >
        <div
          className="image_title_bio"
        >
          <div
            className="profile_image"
          >
            <img
              alt=""
              className="user_pic"
              src="https://picsum.photos/200/300/?random"
            />
          </div>
          <div
            className="title_text_box"
          >
            <p
              className="name"
            />
            <p
              className="user_bio"
            >
              Party: 
            </p>
            <p
              className="user_bio"
            >
              Office of Intrest: 
            </p>
          </div>
        </div>
        <div
          className="admin_decision_on_request"
        >
          <div
            className="main_button"
          >
            <button
              className="btn-accept"
              type="submit"
            >
              Accept
            </button>
            <button
              className="btn-reject"
              type="submit"
            >
              reject
            </button>
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="image_title_bio"
          >
            <div
              className="profile_image"
            >
              <img
                alt=""
                className="user_pic"
                src="https://picsum.photos/200/300/?random"
              />
            </div>
            <div
              className="title_text_box"
            >
              <p
                className="name"
              />
              <p
                className="user_bio"
              >
                Party: 
              </p>
              <p
                className="user_bio"
              >
                Office of Intrest: 
              </p>
            </div>
          </div>,
          <div
            className="admin_decision_on_request"
          >
            <div
              className="main_button"
            >
              <button
                className="btn-accept"
                type="submit"
              >
                Accept
              </button>
              <button
                className="btn-reject"
                type="submit"
              >
                reject
              </button>
            </div>
          </div>,
        ],
        "className": "main-title",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="profile_image"
              >
                <img
                  alt=""
                  className="user_pic"
                  src="https://picsum.photos/200/300/?random"
                />
              </div>,
              <div
                className="title_text_box"
              >
                <p
                  className="name"
                />
                <p
                  className="user_bio"
                >
                  Party: 
                </p>
                <p
                  className="user_bio"
                >
                  Office of Intrest: 
                </p>
              </div>,
            ],
            "className": "image_title_bio",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  className="user_pic"
                  src="https://picsum.photos/200/300/?random"
                />,
                "className": "profile_image",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "user_pic",
                  "src": "https://picsum.photos/200/300/?random",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="name"
                  />,
                  <p
                    className="user_bio"
                  >
                    Party: 
                  </p>,
                  <p
                    className="user_bio"
                  >
                    Office of Intrest: 
                  </p>,
                ],
                "className": "title_text_box",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "name",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Party: ",
                      undefined,
                    ],
                    "className": "user_bio",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Party: ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Office of Intrest: ",
                      undefined,
                    ],
                    "className": "user_bio",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Office of Intrest: ",
                    undefined,
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="main_button"
            >
              <button
                className="btn-accept"
                type="submit"
              >
                Accept
              </button>
              <button
                className="btn-reject"
                type="submit"
              >
                reject
              </button>
            </div>,
            "className": "admin_decision_on_request",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="btn-accept"
                  type="submit"
                >
                  Accept
                </button>,
                <button
                  className="btn-reject"
                  type="submit"
                >
                  reject
                </button>,
              ],
              "className": "main_button",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Accept",
                  "className": "btn-accept",
                  "id": undefined,
                  "onClick": undefined,
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Accept",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "reject",
                  "className": "btn-reject",
                  "id": undefined,
                  "onClick": undefined,
                  "type": "submit",
                },
                "ref": null,
                "rendered": "reject",
                "type": "button",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="main-title"
        >
          <div
            className="image_title_bio"
          >
            <div
              className="profile_image"
            >
              <img
                alt=""
                className="user_pic"
                src="https://picsum.photos/200/300/?random"
              />
            </div>
            <div
              className="title_text_box"
            >
              <p
                className="name"
              />
              <p
                className="user_bio"
              >
                Party: 
              </p>
              <p
                className="user_bio"
              >
                Office of Intrest: 
              </p>
            </div>
          </div>
          <div
            className="admin_decision_on_request"
          >
            <div
              className="main_button"
            >
              <button
                className="btn-accept"
                type="submit"
              >
                Accept
              </button>
              <button
                className="btn-reject"
                type="submit"
              >
                reject
              </button>
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="image_title_bio"
            >
              <div
                className="profile_image"
              >
                <img
                  alt=""
                  className="user_pic"
                  src="https://picsum.photos/200/300/?random"
                />
              </div>
              <div
                className="title_text_box"
              >
                <p
                  className="name"
                />
                <p
                  className="user_bio"
                >
                  Party: 
                </p>
                <p
                  className="user_bio"
                >
                  Office of Intrest: 
                </p>
              </div>
            </div>,
            <div
              className="admin_decision_on_request"
            >
              <div
                className="main_button"
              >
                <button
                  className="btn-accept"
                  type="submit"
                >
                  Accept
                </button>
                <button
                  className="btn-reject"
                  type="submit"
                >
                  reject
                </button>
              </div>
            </div>,
          ],
          "className": "main-title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="profile_image"
                >
                  <img
                    alt=""
                    className="user_pic"
                    src="https://picsum.photos/200/300/?random"
                  />
                </div>,
                <div
                  className="title_text_box"
                >
                  <p
                    className="name"
                  />
                  <p
                    className="user_bio"
                  >
                    Party: 
                  </p>
                  <p
                    className="user_bio"
                  >
                    Office of Intrest: 
                  </p>
                </div>,
              ],
              "className": "image_title_bio",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt=""
                    className="user_pic"
                    src="https://picsum.photos/200/300/?random"
                  />,
                  "className": "profile_image",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "user_pic",
                    "src": "https://picsum.photos/200/300/?random",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="name"
                    />,
                    <p
                      className="user_bio"
                    >
                      Party: 
                    </p>,
                    <p
                      className="user_bio"
                    >
                      Office of Intrest: 
                    </p>,
                  ],
                  "className": "title_text_box",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "name",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Party: ",
                        undefined,
                      ],
                      "className": "user_bio",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Party: ",
                      undefined,
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Office of Intrest: ",
                        undefined,
                      ],
                      "className": "user_bio",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Office of Intrest: ",
                      undefined,
                    ],
                    "type": "p",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="main_button"
              >
                <button
                  className="btn-accept"
                  type="submit"
                >
                  Accept
                </button>
                <button
                  className="btn-reject"
                  type="submit"
                >
                  reject
                </button>
              </div>,
              "className": "admin_decision_on_request",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="btn-accept"
                    type="submit"
                  >
                    Accept
                  </button>,
                  <button
                    className="btn-reject"
                    type="submit"
                  >
                    reject
                  </button>,
                ],
                "className": "main_button",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Accept",
                    "className": "btn-accept",
                    "id": undefined,
                    "onClick": undefined,
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Accept",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "reject",
                    "className": "btn-reject",
                    "id": undefined,
                    "onClick": undefined,
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "reject",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
